lib/iris/tests/test_aggregate_by.py

```diff
@@ -621,18 +1064,73 @@ def test_single_missing(self):
             aggregateby_cube.data, single_expected
         )
 
+    def test_weighted_single_missing(self):
+        # weighted aggregation correctly handles masked data
+        weighted_single_expected = ma.masked_invalid(
+            [
+                [
+                    [0.0, np.nan, 0.0],
+                    [np.nan, 0.0, np.nan],
+                    [0.0, np.nan, 0.0],
+                ],
+                [
+                    [1.0, np.nan, 3.0],
+                    [np.nan, 5.0, np.nan],
+                    [7.0, np.nan, 9.0],
+                ],
+                [
+                    [3.0, np.nan, 9.0],
+                    [np.nan, 15.0, np.nan],
+                    [21.0, np.nan, 27.0],
+                ],
+                [
+                    [7.5, np.nan, 22.5],
+                    [np.nan, 37.5, np.nan],
+                    [52.5, np.nan, 67.5],
+                ],
+                [
+                    [12.0, np.nan, 36.0],
+                    [np.nan, 60.0, np.nan],
+                    [84.0, np.nan, 108.0],
+                ],
+                [
+                    [17.5, np.nan, 52.5],
+                    [np.nan, 87.5, np.nan],
+                    [122.5, np.nan, 157.5],
+                ],
+                [
+                    [24.0, 53.0, 72.0],
+                    [106.0, 120.0, 159.0],
+                    [168.0, 212.0, 216.0],
+                ],
+                [
+                    [31.5, 63.0, 94.5],
+                    [126.0, 157.5, 189.0],
+                    [220.5, 252.0, 283.5],
+                ],
+            ]
+        )
+        aggregateby_cube = self.cube_single_masked.aggregated_by(
+            "height",
+            iris.analysis.MEAN,
+            weights=self.weights_single,
+        )
+
+        # Realize data so that tests with lazy data pass (see
+        # test_lazy_aggregate_by.py).
+        aggregateby_cube.data = as_concrete_data(aggregateby_cube.data)
+        self.assertCML(
+            aggregateby_cube,
+            ("analysis", "aggregated_by", "single_missing.cml"),
+            checksum=False,
+        )
```

Comment:
```suggestion
        self.assertCML(
            aggregateby_cube,
            ("analysis", "aggregated_by", "single_missing.cml"),
        )
```

I _believe_ there are no longer concerns with masks messing with `checksum`, and that we should therefore remove all instances of `checksum=False` in this module. This also removes any need for `as_concrete_data()` since the data gets realised during `assertCML`.